### PR TITLE
liquidation bonus

### PR DESCRIPTION
**Pull request by 0xmtf**

### CODE REVIEW

File: lending_pool/src/router.rs [hunk 1]
```diff
@@ -200,6 +206,21 @@ pub trait RouterModule: proxy::ProxyModule + factory::FactoryModule {
...
+        require!(
+            !self.asset_liquidation_bonus(token_id).is_empty(),
+            "no liquidation_bonus present for asset"
+        );
+
+        let liq_bonus = self.asset_liquidation_bonus(token_id).get();
+        require!(liq_bonus > 0, "liquidation_bonus can not be zero");
+
+        Ok(liq_bonus)
+    }
```
Comment (dorin-iancu) [inline lending_pool/src/router.rs:222]: is_empty is the same as 0, so you're basically checking the same thing twice. Rename and rewrite as:
```suggestion
    fn get_liquidation_bonus_non_zero(
        &self,
        token_id: &TokenIdentifier,
    ) -> SCResult<BigUint> {
        let liq_bonus = self.asset_liquidation_bonus(token_id).get();
        require!(liq_bonus > 0, "no liquidation_bonus present for asset");

        Ok(liq_bonus)
    }
```

File: liquidity_pool/src/liquidity.rs [hunk 2]
```diff
@@ -54,6 +54,37 @@ pub trait LiquidityModule:
...
+    ) -> SCResult<()> {
+        let lend_token_id = self.lend_token().get();
+        require!(
+            payment_token_id == lend_token_id,
+            "lend tokens not supported by this pool"
+        );
+
+        let mut deposit = self.deposit_position(payment_token_nonce).get();
+        require!(
+            &deposit.amount >= &payment_amount,
```
Comment (dorin-iancu) [inline liquidity_pool/src/liquidity.rs:74]: Remove references

File: liquidity_pool/src/liquidity.rs [hunk 3]
```diff
@@ -335,14 +367,29 @@ pub trait LiquidityModule:
 
         self.borrow_position(borrow_position_nonce).clear();
 
+        let bp = self.get_base_precision();
+        let lend_amount_to_return = (asset_amount * (&bp + &liquidation_bonus)) / bp;
+        let lend_tokens = borrow_position.lend_tokens.clone();
```
Comment (dorin-iancu) [inline liquidity_pool/src/liquidity.rs:372]: Use reference instead of cloning
```suggestion
        let lend_tokens = &borrow_position.lend_tokens;
```

**Review by dorin-iancu — CHANGES_REQUESTED** [review]

File: lending_pool/src/router.rs (hunk 1 above)
Comment (mateigraura) [reply lending_pool/src/router.rs]: done

**Review by dorin-iancu — APPROVED** [review]

**Pull request merged**